### ISSUE TITLE
Copy Server Name Error

### ISSUE THREAD

**Issue by samaguire** [DaxStudio/DaxStudio#916]
DAX Studio v3.0.0.725
Connection: PowerBIDesktop - 16.0.39.24

## Summary of Issue
I get an error message in the message window when I click the copy server name button to the clipboard button. Works fine otherwise. This is connected to Power BI Desktop.

![image](https://user-images.githubusercontent.com/62320770/187590364-f55271cc-0f44-4e4b-a3f8-d0fc1bceffb6.png)

## Steps to Reproduce
1.
2.

**Comment by dgosbell** [comment]
Thanks for logging this issue. However, I cannot reproduce it on my machine. 

If you got to Help > About there is a link at the bottom of that dialog to the logs folder. Can you reproduce this issue, then got to the logs folder and look for the file with todays date. You can then either attach that file to this issue or search through it for the text "Error copying server name to clipboard" - there should be a stacktrace and a more detailed error message which you could paste into a new comment on this issue

**Comment by samaguire** [comment]
No problem.

So it seems to copy to the clipboard, as in I can ctrl + v paste it into notepad, however, I've noticed it isn't in the win + v recent clipboard items list. I also have clipboard sharing to my mobile via Phone Link enabled.

``` text
2022-09-01 11:26:56.410 +12:00 [INF] DAX STUDIO VERSION: 3.0.0.725
2022-09-01 11:26:56.449 +12:00 [INF] System Info: OSCaption = Microsoft Windows 11 Pro
2022-09-01 11:26:56.449 +12:00 [INF] System Info: OSRelease = 2009
2022-09-01 11:26:56.449 +12:00 [INF] System Info: OSVersion = 10.0.22000
2022-09-01 11:26:56.449 +12:00 [INF] System Info: OSArchitecture = 64-bit
2022-09-01 11:26:56.449 +12:00 [INF] System Info: VisibleMemoryGB = 15.60
2022-09-01 11:26:56.449 +12:00 [INF] System Info: FreeMemoryGB = 4.18
2022-09-01 11:26:56.449 +12:00 [INF] Culture Info: Name = en-NZ
2022-09-01 11:26:56.454 +12:00 [INF] Culture Info: DisplayName = English (New Zealand)
2022-09-01 11:26:56.454 +12:00 [INF] Culture Info: EnglishName = English (New Zealand)
2022-09-01 11:26:56.454 +12:00 [INF] Culture Info: 2-Letter ISO Name = en
2022-09-01 11:26:56.454 +12:00 [INF] Culture Info: DecimalSeparator = .
2022-09-01 11:26:56.454 +12:00 [INF] Culture Info: GroupSeparator = ,
2022-09-01 11:26:56.454 +12:00 [INF] Culture Info: CurrencySymbol = $
2022-09-01 11:26:56.454 +12:00 [INF] Culture Info: ShortDatePattern = d/MM/yyyy
2022-09-01 11:26:56.455 +12:00 [INF] CommandLine Args: ["C:\\Program Files\\DAX Studio\\DaxStudio.exe","/server=localhost:63712","/database=88b0f842-24e8-4b43-baa1-484294be3ca6"]
2022-09-01 11:26:56.458 +12:00 [INF] Portable Mode: False
2022-09-01 11:26:56.458 +12:00 [INF] Changing minimum log event to Warning
2022-09-01 11:27:39.400 +12:00 [ERR] StatusBarViewModel CopyServerNameToClipboard Error copying server name to clipboard: OpenClipboard Failed (Exception from HRESULT: 0x800401D0 (CLIPBRD_E_CANT_OPEN))
System.Runtime.InteropServices.COMException (0x800401D0): OpenClipboard Failed (Exception from HRESULT: 0x800401D0 (CLIPBRD_E_CANT_OPEN))
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at System.Windows.Clipboard.Flush()
   at System.Windows.Clipboard.CriticalSetDataObject(Object data, Boolean copy)
   at System.Windows.Clipboard.SetDataInternal(String format, Object data)
   at DaxStudio.UI.ViewModels.StatusBarViewModel.CopyServerNameToClipboard()
```

**Comment by dgosbell** [comment]
I've done some searching on that error code and it seems to be a common issue. Apparently only one app can access the clipboard at a time and if you get this error the general advice seems to be to wait a bit and try again. I've added some code to wrap all my clipboard calls in some retry logic which will hopefully fix this issue. This will be in the next release which I hope to get out soon.

**Comment by samaguire** [comment]
👍

**Comment by dgosbell** [comment]
this should now be fixed in the 3.0.1 update
